packages/aws-cdk/lib/api/hotswap/appsync-mapping-templates.ts

```diff
@@ -104,6 +115,15 @@ export async function isHotswappableAppSyncChange(
           if (schemaCreationResponse.status === 'FAILED') {
             throw new Error(schemaCreationResponse.details);
           }
+        } else { //isApiKey
+          if (!sdkRequestObject.id) {
+            // ApiKeyId is optional in CFN but required in SDK. Grab the KeyId from physicalArn if not available as part of CFN template
+            const arnParts = physicalName?.split('/');
+            if (arnParts && arnParts.length === 4) {
+              sdkRequestObject.id = arnParts[3];
+            }
```

Comment:
can we use https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Arn.html here and do `Arn.split` instead?

Reply:
it seems the `Arn.split` only considers one resourceType or resourceName but not "nested" resource such as in this case `arn:aws:appsync:us-east-1:111111111111:apis/apiId/apikeys/api-key-id`
`Arn.split` returns the entire `apiId/apikeys/api-key-id` as the resourceName which then requires a split anyways.

```js
{
    service: 'appsync',
    resource: 'apis',
    partition: 'aws',
    region: 'us-east-1',
    account: '111111111111',
    resourceName: 'apiId/apikeys/api-key-id',
    sep: '/',
    arnFormat: 'arn:aws:service:region:account:resource/resourceName'
}
```
Moreover, `Arn.split` is in `aws-cdk-lib` and the change will require we bump the dependency from dev to runtime which might cause cyclic dependency issue? Let me know what you think